src/main/java/mate/academy/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,29 @@
+package mate.academy.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.dto.user.UserRegistrationRequestDto;
+import mate.academy.dto.user.UserResponseDto;
+import mate.academy.exception.RegistrationException;
+import mate.academy.mapper.UserMapper;
+import mate.academy.repository.user.UserRepository;
+import mate.academy.service.UserService;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto)
+            throws RegistrationException {
+        if (userRepository.findByEmail(requestDto.email()).isPresent()) {
```

Comment:
```suggestion
        if (userRepo.existsByEmail(request.email())) {
```

It's better to use another method to avoid calling isPresent()